packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -1,24 +1,36 @@
 import {
-  OperatorModeState,
-  Stack,
-  StackItem,
-} from '@cardstack/host/components/operator-mode/container';
+  type OperatorModeState,
+  type Stack,
+  type StackItem,
+} from '../components/operator-mode/container';
 import Service from '@ember/service';
 import type CardService from '../services/card-service';
 import { TrackedArray, TrackedObject } from 'tracked-built-ins';
 import { service } from '@ember/service';
-
 import { tracked } from '@glimmer/tracking';
 import { getOwner } from '@ember/application';
 import { scheduleOnce } from '@ember/runloop';
+import stringify from 'safe-stable-stringify';
 import type { Card } from 'https://cardstack.com/base/card-api';
 
 // Below types form a raw POJO representation of operator mode state.
 // This state differs from OperatorModeState in that it only contains cards that have been saved (i.e. have an ID).
 // This is because we don't have a way to serialize a stack configuration of linked cards that have not been saved yet.
 
-type SerializedItem = { card: { id: string }; format: 'isolated' | 'edit' };
-type SerializedStack = { items: SerializedItem[] };
+interface CardItem {
+  type: 'card';
+  id: string;
+  format: 'isolated' | 'edit';
+}
+interface ContainedCardItem {
+  type: 'contained';
+  fieldOfIndex: number; // index of the item in the stack that this is a field of
```

Comment:
somehow `fieldOfIndex` is a weird name. `indexInParent`??

Reply:
I’m trying to say this is a “the index of the item that this is a field of”. Your suggestion is also awkward because the index is of the stack not an index within the parent.

Reply:
I think we actually mean the same thing here